source/agora/test/Base.d

```diff
@@ -1509,8 +1553,17 @@ public APIManager makeTestNetwork (APIManager : TestAPIManager = TestAPIManager,
         test_conf.extra_blocks);
 
     auto net = new APIManager(blocks, test_conf, genesis_start_time);
+    auto name_registry_client = new NameRegistryClient(net.createNameRegistry());
     foreach (ref conf; main_configs)
-        net.createNewNode(conf, file, line);
+    {
+        auto node_api = net.createNewNode(conf, file, line);
+        if(conf.validator.enabled)
+        {
+            string tid_sink;
+            node_api.tid().toString((const(char)[] tid){tid_sink = to!string(tid);});
+            name_registry_client.registerNetworkAddresses(conf.validator.key_pair, [tid_sink]);
+        }
+    }
```

Comment:
Why do you need to link the tid ? And the `registerNetworkAddresses` should probably be done by the validator as part of the startup code, it shouldn't be test specific.

Reply:
During the network tests the nodes communicate through the Localrest, and to get a handle for an other Node(to communicate with it), we need a Tid. There wouldn't  be to much use if we register the (same) IP address for all the test Nodes as we couldn't use the NameRegistry to get a handle for a Node.

Reply:
The code you wrote in `Validator.d` is never executed because `addresses_to_register` was never set in the config in the tests. If it was executed you would see there were exceptions being thrown. The code is not doing what you've intended it to do.

I'll try to give some hints at how this can be improved - but I will need a bit of time. So please stay tuned.